std/algebra/emulated/sw_emulated/point.go

```diff
@@ -822,51 +927,104 @@ func (c *Curve[B, S]) jointScalarMulGLVUnsafe(Q, R *AffinePoint[B], s, t *emulat
 	tableS[2] = c.Add(tableS[2], tableR[0])
 	tableS[3] = c.Neg(tableS[2])
 	f0 := c.baseApi.Mul(&tableS[0].X, c.thirdRootOne)
-	f1 := c.baseApi.Mul(&tableS[1].X, c.thirdRootOne)
 	f2 := c.baseApi.Mul(&tableS[2].X, c.thirdRootOne)
-	f3 := c.baseApi.Mul(&tableS[3].X, c.thirdRootOne)
 	tablePhiS[0] = &AffinePoint[B]{
-		X: *c.baseApi.Lookup2(selector2, selector4, f1, f3, f2, f0),
-		Y: *c.baseApi.Lookup2(selector2, selector4, &tableS[1].Y, &tableS[3].Y, &tableS[2].Y, &tableS[0].Y),
+		X: *c.baseApi.Select(c.api.Xor(selector2, selector4), f2, f0),
+		Y: *c.baseApi.Lookup2(selector2, selector4, &tableS[0].Y, &tableS[2].Y, &tableS[3].Y, &tableS[1].Y),
 	}
 	tablePhiS[1] = c.Neg(tablePhiS[0])
 	tablePhiS[2] = &AffinePoint[B]{
-		X: *c.baseApi.Lookup2(selector2, selector4, f3, f1, f0, f2),
-		Y: *c.baseApi.Lookup2(selector2, selector4, &tableS[3].Y, &tableS[1].Y, &tableS[0].Y, &tableS[2].Y),
+		X: *c.baseApi.Select(c.api.Xor(selector2, selector4), f0, f2),
+		Y: *c.baseApi.Lookup2(selector2, selector4, &tableS[2].Y, &tableS[0].Y, &tableS[1].Y, &tableS[3].Y),
 	}
 	tablePhiS[3] = c.Neg(tablePhiS[2])
 
-	// suppose first bit is 1 and set:
+	// suppose first bits are 1 and set:
 	// Acc = Q + R + Φ(Q) + Φ(R)
 	Acc := c.Add(tableS[1], tablePhiS[1])
-
-	s1 = c.scalarApi.Select(selector1, s1, s3)
-	s2 = c.scalarApi.Select(selector2, s2, s4)
-	t1 = c.scalarApi.Select(selector3, t1, t3)
-	t2 = c.scalarApi.Select(selector4, t2, t4)
+	B1 := Acc
+	// and then add the base point G to it to avoid incomplete additions in the
+	// loop, because when doing doubleAndAdd(Acc, Bi) as (Acc+Bi)+Acc it might
+	// happen that Acc==Bi or -Bi. But now we force Acc to be different than
+	// the stored Bi.  However we need at the end to subtract [2^nbits]G
+	// (harcoded) from the result.
+	//
+	// Acc = Q + R + Φ(Q) + Φ(R) + G
+	Acc = c.Add(Acc, c.Generator())
```

Comment:
Still - in this case still we would hit the incomplete addition when input is `-G`? What about the idea to select between two values to add in the beginning and later to subtract depending if the input is `-G` or not? We could even use `G` and `2G`.